app/celery/tasks.py

```diff
@@ -269,12 +274,13 @@ def send_email(service_id, notification_id, subject, from_address, encrypted_not
                     values=notification.get('personalisation', {})
                 )
 
-                client.send_email(
+                reference = client.send_email(
                     from_address,
                     notification['to'],
                     subject,
                     template.replaced
                 )
+                update_notification_reference_by_id(notification_id, reference)
```

Comment:
Update the notification once the email has been sent to AWS. Wanted to minimse the DB writes here, but I think the update is OK here as we need to save the notification prior to the AWS call, and the ID is only available after,